src/v/debug_bundle/debug_bundle_service.h

```diff
@@ -106,9 +112,46 @@ class service final : public ss::peering_sharded_service<service> {
     ss::future<result<void>> delete_rpk_debug_bundle();
 
 private:
+    /**
+     * @brief Constructs the arguments for the rpk debug bundle command
+     *
+     * @param job_id Job ID
+     * @param params parameters
+     * @return std::vector<ss::sstring> The list of strings to pass to
+     * external_process
+     */
+    std::vector<ss::sstring>
+    build_rpk_arguments(job_id_t job_id, debug_bundle_parameters params);
+
+    /**
+     * @brief Returns the status of the running process
+     *
+     * @return std::optional<debug_bundle_status> Will return std::nullopt if
+     * process was never executed, else will return the debug_bundle_status
+     */
+    std::optional<debug_bundle_status> process_status() const;
+
+    /**
+     * @return Whether or not the RPK debug bundle process is running
+     */
+    bool is_running() const;
+
+private:
+    /// Handler used to emplace stdout/stderr into a buffer
+    struct output_handler;
+    /// Structure used to hold information about the running rpk debug bundle
+    /// process
+    class debug_bundle_process;
     /// Path to the debug bundle directory
     std::filesystem::path _debug_bundle_dir;
     /// Binding called when the rpk path config changes
     config::binding<std::filesystem::path> _rpk_path_binding;
+    /// External process
+    std::unique_ptr<debug_bundle_process> _rpk_process;
+    /// Mutex to guard the initialize method
+    mutex _initialize_mutex;
+    /// Mutex to guard the terminate method
+    mutex _terminate_mutex;
```

Comment:
seems like only one mutex is necessary under the assumption that one cannot terminate until initialization is complete?

Reply:
hmmm yeah you may be right... I need to think a little bit of what that flow would look like but you're probably right :)